vault/resource_kv_secret_v2.go

```diff
@@ -134,8 +138,24 @@ func kvSecretV2Write(ctx context.Context, d *schema.ResourceData, meta interface
 		data[k] = d.Get(k)
 	}
 
-	if _, err := client.Logical().Write(path, data); err != nil {
-		return diag.Errorf("error writing secret data to %s, err=%s", path, err)
+	writeData := func() error {
+		if _, err := client.Logical().Write(path, data); err != nil {
+			if respErr, ok := err.(*api.ResponseError); ok && (respErr.StatusCode == http.StatusBadRequest) {
+				return err
+			} else {
+				return backoff.Permanent(err)
+			}
+		}
+		return nil
+	}
+
+	bo := backoff.WithMaxRetries(backoff.NewConstantBackOff(time.Millisecond*500), 10)
+
+	log.Printf("[DEBUG] creating KVV2 secret %s", path)
+	if err := backoff.RetryNotify(writeData, bo, func(err error, duration time.Duration) {
+		log.Printf("[WARN] create KVV2 %q failed, retrying in %s", path, duration)
+	}); err != nil {
+		return diag.Errorf("error creating KVV2 secret: %s", err)
 	}
```

Comment:
I would prefer to remove the duplicated retry logic here since this resource will only ever support kv-v2. If this becomes an issue in the future we can revisit it then.

```suggestion
	if _, err := client.Logical().Write(path, data); err != nil {
		return diag.Errorf("error writing secret data to %s, err=%s", path, err)
```